book/src/notes.md

```diff
@@ -1,87 +1,67 @@
+##### Note privacy
+The notes are distributed in an encrypted form, and only the owner knows the decryption key. 
+To prove properties of the notes (e.g. ownership) without revealing the notes, we use ZK proofs.
 
-To send a note, the owner needs to prove the ownership which is done by revealing note's nullifier `nf`.  The nullifier is only known to the owner of the note and revealing it destroys the note. All revealed nullifiers are stored in a public nullifier tree `NFtree` to make sure none of the notes are spent twice.
+#### The note commitment tree & the nullifier set
 
-##### Proving ownership in ZK
-All of the notes are kept shielded and created notes are sent to users in an encrypted form. To keep the notes shielded, ZK proofs are used. Using ZK proof, one can prove the ownership without revealing the note itself (see [Action circuit](action.md)).
+Each created note is added to a public Merkle tree. To keep the notes private, instead of the notes themselves, 
+the tree contains note commitments `cm` that are binded to the actual notes. 
+This tree is called a note commitment tree `CMTree`. 
 
-#### The commitment tree
+Notes are never deleted from the note commitment tree, 
+and to signal that the note has been spent and isn't valid anymore, its nullifier `nf` is added to the public nullifier set.
 
-Each created note exists in a public merkle tree `CMtree` of notes. To keep the notes shielded, the tree contains note commitments `cm` instead of the notes themselves. This tree is called a note commitment tree.
+||`cm` isn't published|`cm` is published|
+|-|-|-|
+|`nf` isn't published|note doesn't exist|note created|
+|`nf` is published|impossible|note spent|
 
-#### Note structure
-The full description of the note structure is
+### Note structure fields
+
+Let's look at some fields in the note structure:
 
 ```rust
-pub struct Note<CP: CircuitParameters> {
-	pub owner: User<CP>,
-	pub app: App<CP>,
-	pub value: u64,
-	pub data: CP::CurveScalarField,
-	pub rho: Nullifier<CP>,
-	pub psi: CP::CurveScalarField,
-	pub rcm: CP::CurveScalarField,
+pub struct Note {
+pub value_base: NoteType,
+pub app_data_dynamic: pallas::Base,
+pub value: u64,
+pub nk_com: NullifierKeyCom,
+pub is_merkle_checked: bool,
+...
 }
 ```
+#### The application data
+Notes contain quite a lot of data provided by the application the notes belong to. 
+Part of that data is more long-term (e.g. the application VP) and used to derive the note's `value_base`, while
+the other part is more contextual (e.g. the input parameters for the application VP) and is stored in `app_data_dynamic`. 
+In general, all data that is not used to derive the value base is stored in the `app_data_dynamic`.
 
-, where:
-- `data` is additional information that might be useful to describe notes of a certain type (e.g. NFTs)
-- `rho`  and `psi` are the values used to compute the note's nullifier
-- `rcm` is randomness used to compute the commitment `cm`
+#### Note values
```

Comment:
Maybe it's worth mentioning that value encodes fungibility

Reply:
I'm not sure how it would help. If it is about fungible vs non-fungible tokens, I'm not sure it is true because non-fungible tokens also need to be balanced and to have a value for that, even though carrying a different meaning. If it isn't about it, it feels misleading bc many people would think about the distinction above.